velox/docs/functions/presto/aggregate.rst

```diff
@@ -119,6 +119,44 @@ General Aggregate Functions
     Returns a multimap created from the input ``key`` / ``value`` pairs.
     Each key can be associated with multiple values.
 
+.. function:: reduce_agg(inputValue T, initialState S, inputFunction(S,T,S), combineFunction(S,S,S)) -> S
+
+    Reduces all non-NULL input values into a single value. ``inputFunction``
+    will be invoked for each non-NULL input value. If all inputs are NULL, the
+    result is NULL. In addition to taking the input value, ``inputFunction``
+    takes the current state, initially ``initialState``, and returns the new state.
+    ``combineFunction`` will be invoked to combine two states into a new state.
+    The final state is returned. Throws an error if ``initialState`` is NULL or
+    ``inputFunction`` or ``combineFunction`` returns a NULL.::
+
+        SELECT id, reduce_agg(value, 0, (a, b) -> a + b, (a, b) -> a + b)
```

Comment:
the difference between inputFunction and combineFunction might not be terribly clear to folks reading this at first. Maybe we can have the second example calculating, say, an average, to give a more illustrative example?

Reply:
It is indeed not straightforward. To compute avg, one needs to compute (sum, count) using reduce_agg first, then compute avg from that as a separate projection. Will add this example to the doc.

```
select sum_and_count.sum / sum_and_count.count from (

  select reduce_agg(x, cast(row(0, 0) as row(sum double, count bigint)), 
    (s, x) -> cast(row(s.sum + x, s.count + 1) as row(sum double, count bigint)), 
    (s, s2) -> cast(row(s.sum + s2.sum, s.count + s2.count) as row(sum double, count bigint))) as sum_and_count
  from unnest(array[1,2,3]) as t(x)

);
```